speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp

```diff
@@ -1,75 +1,259 @@
-                                           "show intervention",
-                                           "ask for intervention",
-                                           "intervene when needed",
-                                           "always intervene" };
-static const int s_nrInterventions = sizeof(s_interventionTypes) / sizeof(s_interventionTypes[0]);
-int m_curInterventionTypeIndex = 0;
-
-// GUI label ids
-static int s_interventionTypeId;
 
 // GUI screen handles
 static void* s_scrHandle = NULL;
 static void* s_nextHandle = NULL;
 
-/// @brief Loads the intervention type from the backend config
+// Task
+Task m_task = TASK_NO_TASK;
+
+// Indicators
+bool* m_indicators = new bool[2];
+
+// InterventionType
+InterventionType m_interventionType;
+
+// Environment
+Track track;
+
+// Participant control
+bool* m_pControl = new bool[3];
+
+// Max time
+int m_maxTime = 10.0f;
+int m_maxTimeId;
+
+// User ID
+char* m_userID;
+int m_userIDId;
+
+/// @brief Sets the defaults values
 static void OnActivate(void* /* dummy */)
 {
-    // This whole thing will be deleted in another branch, so for now magic nr 0
-    GfuiLabelSetText(s_scrHandle, s_interventionTypeId, s_interventionTypes[0]);
+    // Set standard Task
+    m_task = TASK_NO_TASK;
+
+    // Set standard indicators
+    bool indicators[] = {false, false};
+    m_indicators = indicators;
+
+    // Set standard interventionType
+    m_interventionType = INTERVENTION_TYPE_NO_SIGNALS;
+
+    // Set standard player control settings
+    bool pControl[] = {false, true, true};
+    m_pControl = pControl;
+
+    // Set standard max time
+    char buf[32];
+    sprintf(buf, "%d", m_maxTime);
+    GfuiEditboxSetString(s_scrHandle, m_maxTimeId, buf);
+
+    // Create random userID
+    std::random_device rd;
+    static std::default_random_engine generator(rd());
+    std::uniform_int_distribution<int> distribution(1, 999999999);
+    sprintf(buf, "%d", distribution(generator));
+
+    // Set default userID
+    m_userID = buf;
+    GfuiEditboxSetString(s_scrHandle, m_userIDId, buf);
 }
 
-/// @brief Saves the settings into the backend config
-static void SaveSettings(void* /* dummy */)
-{
-    SMediator::GetInstance()->SetInterventionType(m_curInterventionTypeIndex);
 
-    // Go to the next screen
-    GfuiScreenActivate(s_nextHandle);
+/// @brief        Sets the task to lane keeping
+/// @param p_info Information on the checkbox
+static void SelectLaneKeeping(tCheckBoxInfo* p_info)
+{
+    if (p_info->bChecked)
+    {
+        m_task = TASK_LANE_KEEPING;
+    }
+    else if (m_task = TASK_LANE_KEEPING)
+    {
+        m_task = TASK_NO_TASK;
+    }
+}
+/// @brief        Sets the task to speed control
+/// @param p_info Information on the checkbox
+static void SelectSpeedControl(tCheckBoxInfo* p_info)
+{
+    if (p_info->bChecked)
+    {
+        m_task = TASK_SPEED_CONTROL;
+    }
+    else if (m_task = TASK_LANE_KEEPING)
+    {
+        m_task = TASK_NO_TASK;
+    }
+}
```

Comment:
This makes it so you can have task lane keeping selected in the menu but the program thinks nothing is selected, to reproduce: Select both, then deselect speed control